### PR TITLE
feat!: use key&value for Labels

### LINKED ISSUE #336: Add YAML schema for func.yaml
We should create a schem for `func.yaml`, this could help with the validation and easier editing of the file in IDEs (values suggestion).

https://developers.redhat.com/blog/2020/11/25/how-to-configure-yaml-schema-to-make-editing-files-easier/

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

# Changes

- :broom: use `key&value` for Labels instead of `name&value` so we are aligned with Kuberentes https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
- :gift:  added `envs` and `labels` validation to the func.yaml schema

Fixes #336

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -42,37 +42,59 @@ func (v Volume) String() string {
...
 	Value *string `yaml:"value"`
 }
 
-func (e Pair) String() string {
+func (e Env) String() string {
 	if e.Name == nil && e.Value != nil {
 		match := regWholeSecret.FindStringSubmatch(*e.Value)
 		if len(match) == 2 {
-			return fmt.Sprintf("All key=value pairs from Secret \"%s\"", match[1])
+			return fmt.Sprintf("Env key=value pairs from Secret \"%s\"", match[1])
```
Comment (lance) [inline config.go:55]: Shouldn't this remain `All`?

File: config.go [hunk 2]
```diff
@@ -42,37 +42,59 @@ func (v Volume) String() string {
...
 		}
 		match = regWholeConfigMap.FindStringSubmatch(*e.Value)
 		if len(match) == 2 {
-			return fmt.Sprintf("All key=value pairs from ConfigMap \"%s\"", match[1])
+			return fmt.Sprintf("Env key=value pairs from ConfigMap \"%s\"", match[1])
```
Comment (lance) [inline config.go:59]: Same as above - shouldn't this be `All`?

File: config.go [hunk 3]
```diff
@@ -42,37 +42,59 @@ func (v Volume) String() string {
...
+	}
+	return ""
+}
+
+type Labels []Label
+type Label struct {
+	// Key consist of optional prefix part (ended by '/') and name part
+	// Prefix part validation pattern: [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*
+	// Name part validation pattern: ([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]
+	Key   *string `yaml:"key" jsonschema:"pattern=^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\\/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$"`
```
Comment (lance) [inline config.go:85]: What I most love about RegEx in general is how easy they are to just take a quick glance at, and know exactly what they are matching. :sob: :rofl:

**Review by lance — COMMENTED** [review]
Just one question about the console output.

File: config.go (hunk 1 above)
Comment (zroubalik) [reply config.go:55]: Good catch! copypaste error 🤦

File: config.go (hunk 3 above)
Comment (zroubalik) [reply config.go:85]: lol 🔥

File: config.go (hunk 1 above)
Comment (zroubalik) [reply config.go]: Fixed

File: config.go (hunk 3 above)
Comment (zroubalik) [reply config.go:85]: I copypasted those two regexes directly from Kubernetes validation code. I tested the combined regex, so it should be ok 🤞

**Pull request merged**